general-superstaq/general_superstaq/superstaq_exceptions.py

```diff
@@ -61,4 +71,19 @@ def __init__(self, job_id: str, status: str) -> None:
             job_id: The job identifier of the unsuccessful job.
             status: The status of the unsuccessful job.
         """
-        super().__init__(f"Job {job_id} terminated with status {status}.")
+        super().__init__(f"Job {job_id} terminated with status {status}")
+
+
+class SuperstaqServerException(SuperstaqException):
```

Comment:
also @vtomole do you remember why we called these e.g. `SuperstaqException` instead of `SuperstaqError` (which would be more in keeping with most other python errors)?

Reply:
> wdyt about making *any* error with a status code a SuperstaqServerException? 

i think this sounds good. 

>  this should also probably be where we put the "contact us..." message

so something like
```python
err_msg = f"{message}. (Status code: {status_msg}). Contact us ..."
```
?

Reply:
> so something like

made this changes. also moved the "Non-retriable error making request to Superstaq API" message to the end. changes would also need to be made in server so we raise `SuperstaqServerException` instead of `SuperstaqException` for the integration tests to pass.

Reply:
> do you remember why we called these e.g. SuperstaqException instead of SuperstaqError 

I don't have a better answer other than[ "cirq-ionq did it"](https://github.com/quantumlib/Cirq/blob/8e70f775f4f4f5a4658027d8a7124d308aa3bd3c/cirq-ionq/cirq_ionq/ionq_exceptions.py#L21) :(

Reply:
> so something like
> 
> ```python
> err_msg = f"{message}. (Status code: {status_msg}). Contact us ..."
> ```
> 
> ?

right now when a superstaq exception is raised by the server it always looks something like:
```
SuperstaqException: Status code: 500, Message: 'Non-retriable error making request to Superstaq API, Internal Server Error.

If you would like to contact a member of our team, email us at superstaq@infleqtion.com or join our Slack workspace: https://join.slack.com/t/superstaq/shared_invite/zt-1wr6eok5j-fMwB7dPEWGG~5S474xGhxw'
```
the second piece is currently just part of the message returned from the server, but i was wondering if maybe it makes sense to hard code it in `SuperstaqServerException` instead?

Reply:
> but i was wondering if maybe it makes sense to hard code it in SuperstaqServerException instead?

But do we want this contact us message to appear for every error that's raised by the server? So for example, if they ask for more shots than are available in the device, I don't think it makes sense for the contact us message to appear. Or do we hardcode it only for 400 (and others I might not be aware of) codes?

Reply:
ah my bad - i thought this message was coming from the server but it looks like it's already hardcoded in `_SuperstaqClient._handle_status_codes`

but in that case: the status code ends up in a weird place in the current message, e.g.
```
SuperstaqServerException: Too many shots.

If you would like to contact a member of our team, email us at superstaq@infleqtion.com or join our Slack workspace: https://join.slack.com/t/superstaq/shared_invite/zt-1wr6eok5j-fMwB7dPEWGG~5S474xGhxw (Status code: 400, non-retriable error making request to Superstaq API)
```
do we want to change this to put it before the contact us part? e.g.
```
SuperstaqServerException: Too many shots. (Status code: 400, non-retriable error making request to Superstaq API)

If you would like to contact a member of our team, email us at superstaq@infleqtion.com or join our Slack workspace: https://join.slack.com/t/superstaq/shared_invite/zt-1wr6eok5j-fMwB7dPEWGG~5S474xGhxw
```
(probably requires adding an optional third parameter for `SuperstaqServerException`)

alternatively we could just add a couple newlines after the url in that particular message to make it:
```
SuperstaqServerException: Too many shots.

If you would like to contact a member of our team, email us at superstaq@infleqtion.com or join our Slack workspace: https://join.slack.com/t/superstaq/shared_invite/zt-1wr6eok5j-fMwB7dPEWGG~5S474xGhxw

(Status code: 400, non-retriable error making request to Superstaq API)
```

Reply:
made changes agreed irl